examples/runners/wgpu/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+    // find an adapter that can render to this surface.
+    let mut surface = if !cfg!(target_os = "android") {
+        Some(unsafe { instance.create_surface(&window) })
+    } else {
+        None
+    };
+
+    let adapter = instance
+        .request_adapter(&wgpu::RequestAdapterOptions {
+            power_preference: wgpu::PowerPreference::default(),
+            // Request an adapter which can render to our surface
+            compatible_surface: surface.as_ref(),
+        })
+        .await
+        .expect("Failed to find an appropriate adapter");
+
+    // Create the logical device and command queue
+    let (device, queue) = adapter
+        .request_device(
+            &wgpu::DeviceDescriptor {
+                features: wgpu::Features::empty(),
+                limits: wgpu::Limits::default(),
+                shader_validation: true,
+            },
+            None,
+        )
+        .await
+        .expect("Failed to create device");
+
+    // Load the shaders from disk
+    let module = device.create_shader_module(wgpu::include_spirv!(env!("sky_shader.spv")));
+
+    let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+        label: None,
+        bind_group_layouts: &[],
+        push_constant_ranges: &[],
+    });
+
+    let render_pipeline = device.create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+        label: None,
+        layout: Some(&pipeline_layout),
+        vertex_stage: wgpu::ProgrammableStageDescriptor {
+            module: &module,
+            entry_point: "main_vs",
+        },
+        fragment_stage: Some(wgpu::ProgrammableStageDescriptor {
+            module: &module,
+            entry_point: "main_fs",
+        }),
+        // Use the default rasterizer state: no culling, no depth bias
+        rasterization_state: None,
+        primitive_topology: wgpu::PrimitiveTopology::TriangleList,
+        color_states: &[swapchain_format.into()],
+        depth_stencil_state: None,
+        vertex_state: wgpu::VertexStateDescriptor {
+            index_format: wgpu::IndexFormat::Uint16,
+            vertex_buffers: &[],
+        },
+        sample_count: 1,
+        sample_mask: !0,
+        alpha_to_coverage_enabled: false,
+    });
+
+    let mut sc_desc = wgpu::SwapChainDescriptor {
+        usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+        format: swapchain_format,
+        width: size.width,
+        height: size.height,
+        present_mode: wgpu::PresentMode::Mailbox,
+    };
+
+    let mut swap_chain = surface
+        .as_ref()
+        .map(|surface| device.create_swap_chain(&surface, &sc_desc));
+
+    event_loop.run(move |event, _, control_flow| {
+        // Have the closure take ownership of the resources.
+        // `event_loop.run` never returns, therefore we must do this to ensure
+        // the resources are properly cleaned up.
+        let _ = (&instance, &adapter, &module, &pipeline_layout);
+
+        *control_flow = ControlFlow::Wait;
```

Comment:
I have no idea what practical differences are (just what the raw functionality does), is there a reason you changed this from Poll to Wait?

Reply:
That's explained in the relevant commit making this change: 54752c289c7d1b3a48488db3278344fddffcab0f

> The image currently does not change and the OS will notify us when to
> redraw (ie. after window resizes). This is going to save power
> especially on mobile devices.

> As soon as interactive or animating visuals are introduced to this
> example redraws can be requested with `window.request_redraw()`.